src/BuildInfo.ts

```diff
@@ -24,25 +24,32 @@ export class BuildInfo {
     public static readonly UNKNOWN_VERSION_ID = -1;
     private static readonly MAJOR_VERSION_MULTIPLIER = 10000;
     private static readonly MINOR_VERSION_MULTIPLIER = 100;
-    private static readonly PATTERN = /^([\d]+)\.([\d]+)(?:\.([\d]+))?(-[\w]+)?(-SNAPSHOT)?(-BETA-.)?$/;
 
     public static calculateServerVersionFromString(versionString: string): number {
         if (versionString == null) {
             return BuildInfo.UNKNOWN_VERSION_ID;
         }
-        const info = BuildInfo.PATTERN.exec(versionString);
-        if (info === null) {
+        const mainParts = versionString.split('-');
+        const tokens = mainParts[0].split('.');
+
+        if (tokens.length < 2) {
             return BuildInfo.UNKNOWN_VERSION_ID;
         }
-        const major = Number.parseInt(info[1]);
-        const minor = Number.parseInt(info[2]);
+
+        const major = +tokens[0];
+        const minor = +tokens[1];
         let patch: number;
-        if (info[3] === undefined) {
+        if (tokens.length === 2) {
             patch = 0;
         } else {
-            patch = Number.parseInt(info[3]);
+            patch = +tokens[2];
+        }
```

Comment:
These lines could be simplified as:
```
let patch = (tokens.length === 2)? 0 : +tokens[2];
```

Reply:
done in 672546d6

also changed return part in the same way and added a comment